### PR TITLE
DDF-4107, DDF-4106 Adds prettier formatter to Intrigue

### PR DESCRIPTION
**Pull request by andrewkfiedler**
#### What does this PR do?
- Adds prettier formatter to Intrigue
- Migrates source components

#### Who is reviewing it? 
@Schachte 
@brendan-hofmann
@rzwiefel
@tbatie
@vinamartin
@djblue 

#### Note to Reviewers
Check out the prettier config and tell me if you disagree 😄 
```
{
    singleQuote: true,
    trailingComma: 'es5'
}
```
I put some files in the ignore because prettier said formatting them would result in possible behavior changes.

#### How should this be tested?
To test the source migration make a few sources and verify it works as before.  When sources go down, things should update in the UI.  Make a source action and verify it shows up in the UI and works as intended.

To test the prettier formatter, pull down the changes.  Run `yarn build` and verify it passes.  Make a change to a file that goes against the formatter.  Run `yarn build` and verify it fails.

Right now I only have it setup to verify changes versus master for building, but we could swap it to look at all files.

#### What are the relevant tickets?
https://codice.atlassian.net/projects/DDF/issues/DDF-4106
https://codice.atlassian.net/projects/DDF/issues/DDF-4107

screen shot 2018-08-29 at 9 53 46 pm

### CODE REVIEW

**Comment by andrewkfiedler** [comment]
@AzGoalie or @brendan-hofmann let me know if you have some time to test these new scripts on windows, I don't want to break windows devs.

**Comment by ethantmanns** [comment]
I verified this works on Windows 👍

**Comment by andrewkfiedler** [comment]
build now

File: ui/packages/ace/lib/format.js [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+  function error(message) {
+    errors = true;
+    console.error(color.red(`\n${message}`));
+  }
+
+  function tellUserHowToFormat() {
+    console.error(
+      '\n' +
+        color.red(
```
Comment (brendan-hofmann) [inline ui/packages/ace/lib/format.js]: ❓ This red is pretty hard to read against powershell blue, is there a lighter shade of red we could use?

Comment (andrewkfiedler) [reply ui/packages/ace/lib/format.js]: Choose your color: https://github.com/doowb/ansi-colors

**Comment by andrewkfiedler** [comment]
@brendan-hofmann yeah I think I need to exclude those files because I think they get generated during the build each time.

**Comment by andrewkfiedler** [comment]
build now

**Comment by andrewkfiedler** [comment]
build now

**Comment by andrewkfiedler** [comment]
build now

**Comment by andrewkfiedler** [comment]
build now

**Pull request merged**